lib/utils/import.js

```diff
@@ -9,11 +9,59 @@ const {
 } = require('../utils/types');
 
 module.exports = {
+  gatherImports,
+  getImportFromMap,
   getSourceModuleNameForIdentifier,
   getSourceModuleName,
   getImportIdentifier,
 };
 
+const IMPORTS_TO_GATHER = [
```

Comment:
I think we should gather all the imports, instead of just these. It would make it a lot easier to reuse this logic for arbitrary rules, instead of just these. For example, we have things that check if something is in a `Component`, but I don't think we currently check if they imported it with a different name.